File: src/dataset/javascript/kindergarten-garden_exercism-javascript-ThomasZumsteg/gemini/GeminiFlash_kindergarten-garden_v4.js
var DEFAULT_STUDENTS = ['Alice', 'Bob', 'Charlie', 'David', 'Eve',
    'Fred', 'Ginny', 'Harriet', 'Ileana', 'Joseph', 'Kincaid', 'Larry'];

var Garden = function (garden, students) {
    students = students ? [...students].sort() : [...DEFAULT_STUDENTS];
    const gardenRows = garden.split('\n');
    const plantMap = { 'C': 'clover', 'G': 'grass', 'R': 'radishes', 'V': 'violets' };

    for (let i = 0; i < students.length; i++) {
        const student = students[i].toLowerCase();
        const plants = [];
        for (let j = 0; j < gardenRows.length; j++) {
            const row = gardenRows[j];
            plants.push(plantMap[row[2 * i]]);
            plants.push(plantMap[row[2 * i + 1]]);
        }
        this[student] = plants;
    }
};

export default Garden;

File: src/dataset/python/kindergarten-garden/gemini/GeminiFlash_kindergarten-garden_v4.py
plant = {'V':'Violets', 'R':'Radishes', 'G':'Grass', 'C':'Clover'}
stu = ['Alice', 'Bob', 'Charlie', 'David', 'Eve', 'Fred', 'Ginny', 'Harriet', 'Ileana', 'Joseph', 'Kincaid', 'Larry']
class Garden: 
    def __init__(self, diagram, students=stu):
        self.diagram = diagram
        self.students = sorted(students)
        self.plant_rows = diagram.split('\n')

    def plants(self, studentx):
        try:
            si = self.students.index(studentx) * 2
        except ValueError:
            return []

        stu_plants = [row[si:si+2] for row in self.plant_rows]
        return [plant[p] for row in stu_plants for p in row]